clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -18347,7 +18347,7 @@ void Sema::SetFunctionBodyKind(Decl *D, SourceLocation Loc, FnBodyKind BodyKind,
   }
 }
 
-bool Sema::CheckOverridingFunctionAttributes(const CXXMethodDecl *New,
+bool Sema::CheckOverridingFunctionAttributes(CXXMethodDecl *New,
```

Comment:
It seems possibly strange to me that a function called "Check" would mutate a value passed into it. I would expect it to be passive.

Can the mutation come later, or in a different function, or perhaps this should be renamed?

Reply:
Afaik there are a bunch of `Check...` functions that mutate their arguments, so this doesn’t seem too strange to me—that said, if this function never actually mutates anything, then I’m fine w/ adding the `const` to document that.

Reply:
I did have to make this new Decl mutable, because the verification can implicitly propagate (certain) effects from the base virtual method to its overrides. (Originally I had done this with an ugly `const_cast` but we discussed this and chose instead to make the parameter mutable.)

Reply:
Works for me!